### ISSUE TITLE
Drop @huggingface/transformers dependency  use onnxruntime-node directly

### ISSUE THREAD

**Issue by shsolomo** [shsolomo/myelin#55]
## Problem

`@huggingface/transformers@3.8.1` has hard dependencies on `onnxruntime-web` and `sharp`  browser/image packages that myelin (a Node.js CLI tool) never uses. This causes real install failures:

- **Windows global install**: `onnxruntime-web` contains WebGPU symlinks that fail tar extraction, cascading into `ENOENT spawn cmd.exe` for all native module builds
- **npm overrides don't apply during `npm install -g`**: Our `package.json` override that aliases `onnxruntime-web` to `onnxruntime-node` is ignored because npm only honors overrides from the root project
- **Beta testers hit this immediately**: Ian had to manually add overrides to his global npm config to install

### Current workaround (shipped in 37da333)

Users must add the override to their global npm prefix `package.json`:

```json
{
  "overrides": {
    "onnxruntime-web": "npm:onnxruntime-node@>=1.21.0"
  }
}
```

This is documented in INSTALL.md but is friction we shouldn't impose.

## Proper Fix

Drop `@huggingface/transformers` entirely. We already depend on `onnxruntime-node` directly. HF transformers is a convenience wrapper around model loading + tokenization  both of which we can implement with ~200-300 lines of focused code.

### What we use today (2 files, 5 API calls)

| API | File | Purpose |
|-----|------|---------|
| `pipeline("feature-extraction", model)` | `embeddings.ts` | Load all-MiniLM-L6-v2 ONNX model |
| `pipeline(text, {pooling, normalize})` | `embeddings.ts` | Run inference, 384-dim embeddings |
| `AutoTokenizer.from_pretrained(dir)` | `ner.ts` | Load GLiNER DeBERTa v2 tokenizer from disk |
| `tokenizer.encode(text, opts)` | `ner.ts` | Tokenize text to IDs |
| `tokenizer.model.tokens_to_ids` | `ner.ts` | Access vocabulary map |

### What replaces them

**Embeddings (all-MiniLM-L6-v2  BERT/WordPiece tokenizer):**
1. Load `tokenizer.json` from the model cache directory  it's self-describing (contains vocab, algorithm, normalization, pre/post-processing config)
2. Implement WordPiece tokenization: normalize, pre-tokenize, split, lookup in vocab, add special tokens (`[CLS]`, `[SEP]`)
3. Load the ONNX model directly via `onnxruntime-node` (`InferenceSession.create()`)
4. Run inference: construct `input_ids`, `attention_mask`, `token_type_ids` tensors, then `session.run()`
5. Mean pooling + L2 normalization on the output (trivial  already understood)

**NER (GLiNER DeBERTa v2  SentencePiece tokenizer):**
1. Load `tokenizer.json` from `~/.cache/myelin/models/gliner/`  already downloaded
2. Implement SentencePiece/Unigram tokenization: normalize, encode using the Unigram model in tokenizer.json
3. Build the vocabulary map from tokenizer.json `model.vocab`
4. `encode()` function: text to token IDs (no special tokens, matching current usage)

### Dependencies eliminated

| Package | Size | Why it's dead weight |
|---------|------|---------------------|
| `@huggingface/transformers` | ~15MB | Polyglot ML framework, we use less than 1% |
| `onnxruntime-web` | ~40MB | Browser WASM/WebGPU runtime  breaks Windows installs |
| `sharp` | ~30MB native | Image processing  myelin is text-only |
| `@huggingface/jinja` | ~200KB | Template engine for chat templates |
| + transitive deps | varies | global-agent, type-fest, etc. |

### Implementation approach

1. Create `src/memory/tokenizer.ts`  generic tokenizer loader that reads `tokenizer.json` format
2. Implement WordPiece algorithm (well-documented, ~100 lines)
3. Implement Unigram/SentencePiece algorithm (~100 lines)
4. Refactor `embeddings.ts` to use onnxruntime-node `InferenceSession` directly
5. Refactor `ner.ts` to use the new tokenizer module
6. Remove `@huggingface/transformers` from dependencies
7. Remove the `overrides` field from package.json (no longer needed)
8. Update INSTALL.md to remove the workaround section

### Risks

- **Tokenizer edge cases**: HF's tokenizer handles Unicode normalization, pre-tokenization rules, and edge cases we'd need to replicate. The `tokenizer.json` format is well-specified but non-trivial.
- **Model loading**: Currently HF handles ONNX model download + caching. We already handle this for GLiNER (`scripts/download-models.mjs`), so extending to embeddings is straightforward.
- **Testing**: Need comprehensive tokenizer tests comparing our output to HF's output for the same inputs.

### Success criteria

- `npm install -g github:shsolomo/myelin` works on Windows without any overrides
- No `onnxruntime-web` or `sharp` in the dependency tree
- Embedding + NER output identical to current implementation (regression tests)
- Install size reduced significantly

**Comment by shsolomo** [comment]
Shipped. @huggingface/transformers removed, custom tokenizer (WordPiece + Unigram), direct ONNX inference for embeddings. 543 tests pass.
